streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryTimeOrderedKeyValueBuffer.java

```diff
@@ -339,15 +301,18 @@ private void restoreBatch(final Collection<ConsumerRecord<byte[], byte[]>> batch
                     cleanPut(
                         time,
                         key,
-                        new ContextualRecord(
-                            value,
-                            new ProcessorRecordContext(
-                                record.timestamp(),
-                                record.offset(),
-                                record.partition(),
-                                record.topic(),
-                                record.headers()
-                            )
+                        new BufferValue(
+                            new ContextualRecord(
+                                value,
+                                new ProcessorRecordContext(
+                                    record.timestamp(),
+                                    record.offset(),
+                                    record.partition(),
+                                    record.topic(),
+                                    record.headers()
+                                )
+                            ),
+                            null
```

Comment:
Not sure the best default "prior" value... `null` removes it from the view, so that might not be great...
Could also default to the "old" value, but we'd have to deserialize it...

Reply:
Much as it sucks to deserialize it does seem like the "correct" thing to do here (and below). That said maybe I'm misunderstanding you because I don't see why you have to deserialize anything..?

Reply:
After reflection, I agree with you. I'll make the change, and then we can discuss further.